### PR TITLE
Bugfix/pkg import paths

### PR DESCRIPTION
**Pull request by creisle**
Saw this error running the lastest version (0.0.3)

```
Traceback (most recent call last):
  File "src/feverous/baseline/predictor/train_verdict_predictor.py", line 19, in <module>
    from utils.annotation_processor import AnnotationProcessor, EvidenceType
ModuleNotFoundError: No module named 'utils'
```

This PR fixes the above path bugs. Now that the src code is being packed under a feverous folder, any global imports should start with "feverous." so that python can find the submodules etc. 

There's a mix of global and relative path imports in the code base so I wasn't sure which one you wanted, I've gone with global imports for now since the scripts can still be run directly then

Note: This fix is required for the other PR (https://github.com/Raldir/FEVEROUS/pull/17) which I have put into draft pending this PR

### CODE REVIEW

**Comment by Raldir** [comment]
Oh no, very sorry for that - should not have happened. I'll look into that asap. Thank you very much Caralyn for the efforts!

**Comment by Raldir** [comment]
Hi Caralyn. I do not think adjusting the path is needed as long as the correct PYTHONPATH is set. I cannot see a line for this in your snakemake but I might be missing something. Using the code as shown in the README works fine on my end. Could you double-check and let me know?

**Comment by creisle** [comment]
Adding the PYTHONPATH setting before each command should work if the user has the code locally and doesn't install from pip. However, the motivation for me to develop the packaging step originally was to avoid needing to manipulate the path/pythonpath environment variable(s). I haven't come across any other python packages that require setting the PYTHONPATH. The above PR changes means you won't need the PYTHONPATH step and it should work both locally and as a package install from pip

**Comment by Raldir** [comment]
I see. do you mind sharing the command you are trying to use that is failing? Running `python -m feverous.baseline.predictor.evaluate_verdict_predictor --input_path data/dev.combined.not_precomputed.p5.s5.t3.cells.jsonl --wiki_path data/feverous_wikiv1.db --model_path models/feverous_verdict_predictor` seems to work fine.

**Comment by creisle** [comment]
Thanks for trying that! Looking at your steps vs mine there seem to be 2 key differences

1. conda vs regular python
2. installing from remote pip (`pip install feverous`) vs installing locally (`pip install .`) 

I can try with the latter change right away, however I can't use this for testing development changes (ex. snakemake) since it has to be publish remotely first. It **should** be identical to installing locally but maybe something strange is going on. I will try that one out and diff the code version it installs if it works

I don't have conda set up here as I have had a lot of troubles with it in the past, the most problematic being it caused a number of issues in unrelated development projects since it likes to globally manipulate environments/paths. So I'll try the remote pip install with python first if that's ok and then if it still has the error I'll look into conda

**Comment by creisle** [comment]
Good news and weird news!

The problem is with installing via pypi vs local, however....

## TL;DR. Summary

- The local install using pip does not match the install from pypi. I am not sure how this is possible but its probably beyond my python expertise. I am wondering if the setup.py file used to publish the package to pypi was altered at all? or maybe having PYTHONPATH set when publishing made a difference
- The subpackages are being installed twice, top-level and under feverous (when installing from pypi) which why we don't see the error when installing from pypi but we do when installing locally (`pip install .`)

## Details

I took a look at what it is doing. If I cd into site-packages post-install I can see that it is installing feverous as one package, but also installing all of its submodules as top level modules which is what is getting around the import issues.

```text
btree feverous baseline utils database examples
feverous
|-- baseline
|   |-- drqa
|   |   |-- __init__.py
|   |   |-- retriever
|   |   |   |-- doc_db.py
|   |   |   |-- __init__.py
|   |   |   |-- simple.py
|   |   |   |-- tfidf_doc_ranker.py
|   |   |   `-- utils.py
|   |   `-- tokenizers
|   |       |-- corenlp_tokenizer.py
|   |       |-- __init__.py
|   |       |-- regexp_tokenizer.py
|   |       |-- simple_tokenizer.py
|   |       |-- spacy_tokenizer.py
|   |       `-- tokenizer.py
|   |-- drqascripts
|   |   |-- build_tfidf_lines.py
|   |   |-- build_tfidf.py
|   |   `-- __init__.py
|   |-- __init__.py
|   |-- predictor
|   |   |-- evaluate_verdict_predictor.py
|   |   |-- __init__.py
|   |   `-- train_verdict_predictor.py
|   `-- retriever
|       |-- build_db.py
|       |-- build_tfidf.py
|       |-- combine_retrieval.py
|       |-- document_entity_tfidf_ir.py
|       |-- eval_combined_retriever.py
|       |-- eval_doc_retriever.py
|       |-- eval_sentence_retriever.py
|       |-- eval_tab_retriever.py
|       |-- __init__.py
|       |-- predict_cells_from_table.py
|       |-- sentence_tfidf_drqa.py
|       |-- table_tfidf_drqa.py
|       `-- train_cell_evidence_retriever.py
|-- database
|   |-- create_wiki_db.py
|   |-- feverous_db.py
|   |-- __init__.py
|   `-- utils.py
|-- evaluation
|   |-- evaluate.py
|   |-- feverous_scorer.py
|   |-- __init__.py
|   `-- prepare_submission.py
|-- examples
|   |-- __init__.py
|   |-- read_context.py
|   |-- read_lists.py
|   `-- read_tables.py
|-- __init__.py
`-- utils
    |-- annotation_processor.py
    |-- __init__.py
    |-- log_helper.py
    |-- prepare_model_input.py
    |-- util.py
    |-- wiki_list.py
    |-- wiki_page.py
    |-- wiki_processor.py
    |-- wiki_section.py
    |-- wiki_sentence.py
    `-- wiki_table.py

baseline
|-- drqa
|   |-- __init__.py
|   |-- retriever
|   |   |-- doc_db.py
|   |   |-- __init__.py
|   |   |-- simple.py
|   |   |-- tfidf_doc_ranker.py
|   |   `-- utils.py
|   `-- tokenizers
|       |-- corenlp_tokenizer.py
|       |-- __init__.py
|       |-- regexp_tokenizer.py
|       |-- simple_tokenizer.py
|       |-- spacy_tokenizer.py
|       `-- tokenizer.py
|-- drqascripts
|   |-- build_tfidf_lines.py
|   |-- build_tfidf.py
|   `-- __init__.py
|-- __init__.py
|-- predictor
|   |-- evaluate_verdict_predictor.py
|   |-- __init__.py
|   `-- train_verdict_predictor.py
`-- retriever
    |-- build_db.py
    |-- build_tfidf.py
    |-- combine_retrieval.py
    |-- document_entity_tfidf_ir.py
    |-- eval_combined_retriever.py
    |-- eval_doc_retriever.py
    |-- eval_sentence_retriever.py
    |-- eval_tab_retriever.py
    |-- __init__.py
    |-- predict_cells_from_table.py
    |-- sentence_tfidf_drqa.py
    |-- table_tfidf_drqa.py
    `-- train_cell_evidence_retriever.py

utils
|-- annotation_processor.py
|-- __init__.py
|-- log_helper.py
|-- prepare_model_input.py
|-- util.py
|-- wiki_list.py
|-- wiki_page.py
|-- wiki_processor.py
|-- wiki_section.py
|-- wiki_sentence.py
`-- wiki_table.py

database
|-- create_wiki_db.py
|-- feverous_db.py
|-- __init__.py
`-- utils.py

examples
|-- __init__.py
|-- read_context.py
|-- read_lists.py
`-- read_tables.py
```

This means there is 2 copies of the files for all the submodules being installed.

We're pretty deep in the weeds on how python import/installs work now but my best guess is that pip install is somehow magically figuring out that it needs to put the submodules as global modules as well. What I can't figure out is why it doesn't do the same thing locally (same tree command from local install).

```text
feverous
|-- baseline
|   |-- drqa
|   |   |-- __init__.py
|   |   |-- retriever
|   |   |   |-- doc_db.py
|   |   |   |-- __init__.py
|   |   |   |-- simple.py
|   |   |   |-- tfidf_doc_ranker.py
|   |   |   `-- utils.py
|   |   `-- tokenizers
|   |       |-- corenlp_tokenizer.py
|   |       |-- __init__.py
|   |       |-- regexp_tokenizer.py
|   |       |-- simple_tokenizer.py
|   |       |-- spacy_tokenizer.py
|   |       `-- tokenizer.py
|   |-- drqascripts
|   |   |-- build_tfidf_lines.py
|   |   |-- build_tfidf.py
|   |   `-- __init__.py
|   |-- __init__.py
|   |-- predictor
|   |   |-- evaluate_verdict_predictor.py
|   |   |-- __init__.py
|   |   `-- train_verdict_predictor.py
|   `-- retriever
|       |-- build_db.py
|       |-- build_tfidf.py
|       |-- combine_retrieval.py
|       |-- document_entity_tfidf_ir.py
|       |-- eval_combined_retriever.py
|       |-- eval_doc_retriever.py
|       |-- eval_sentence_retriever.py
|       |-- eval_tab_retriever.py
|       |-- __init__.py
|       |-- predict_cells_from_table.py
|       |-- sentence_tfidf_drqa.py
|       |-- table_tfidf_drqa.py
|       `-- train_cell_evidence_retriever.py
|-- database
|   |-- create_wiki_db.py
|   |-- feverous_db.py
|   |-- __init__.py
|   `-- utils.py
|-- evaluation
|   |-- evaluate.py
|   |-- feverous_scorer.py
|   |-- __init__.py
|   `-- prepare_submission.py
|-- examples
|   |-- __init__.py
|   |-- read_context.py
|   |-- read_lists.py
|   `-- read_tables.py
|-- __init__.py
`-- utils
    |-- annotation_processor.py
    |-- __init__.py
    |-- log_helper.py
    |-- prepare_model_input.py
    |-- util.py
    |-- wiki_list.py
    |-- wiki_page.py
    |-- wiki_processor.py
    |-- wiki_section.py
    |-- wiki_sentence.py
    `-- wiki_table.py
```

There may be some downsides to having the submodules under feverous as well as top-level, they have pretty generic names so it might end up conflicting with other packages eventually.

Unrelated and not a big deal (just makes the package size a little bigger than it needs to be) but it might be good to add the pyc/pycache files to the ignore list in the MANIFEST.in

When I go to look record file (`venv3.7_ubuntu/lib/python3.7/site-packages/feverous-0.0.3.dist-info/RECORD`) for the pypi pip installed version it shows a bunch of cache files were copied but also confirms that the submodules were installed at top level and under the main package.
